ovs-update.md

```diff
@@ -0,0 +1,281 @@
+disruption.
+
+A new Open vSwitch for podified OVN gateways is shipped as a new container
+image. During the image update, the DaemonSet will be updated to use the new
+image. The DaemonSet update will cause the old pods to be terminated and new
+pods to be started. This will necessarily restart the `ovs-vswitchd` processes.
+
+Right now, there is no provision to mitigate the effect of this restart.
+Meaning, the new process will start with empty OpenFlow tables and will
+invalidate datapath flows. It will wait for `ovn-controller` to reconnect and
+reconfigure it via OpenFlow control socket. In the meantime, the traffic may be
+disrupted.
+
+### Proposal
+
+To mitigate the effect of `ovs-vswitchd` restart, we can utilize
+`other_config:flow-restore-wait` mechanism in Open vSwitch.
+[This mechanism](https://github.com/openvswitch/ovs/blob/fbade819d2de1685c49e1deaf62d45049a5c2a27/vswitchd/vswitch.xml#L104)
+allows Open vSwitch to restore the OpenFlow tables from a backup file after a
+restart.
+
+This mechanism is already utilized in the Open vSwitch RPM package when doing
+`systemctl reload openvswitch`. Since systemd unit files are not used to manage
+`ovs-vswitchd` in pods, we should implement this mechanism in shutdown and
+startup scripts used for OVS DaemonSet.[^0]
+
+The proposed implementation would work as follows:
+
+- Before shutting down `ovs-vswitchd`,
+  [`PreStop` hook](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/)
+  for OVS DaemonSet pods would set `other_config:flow-restore-wait` to `true`
+  in the OVSDB. This will indicate to the new `ovs-vswitchd` process that it
+  should not flush the kernel datapath or connect to its controllers until
+  saved flows are restored.
+
+- In addition, the `PreStop` hook would save the OpenFlow tables to a file in
+  the pod's volume. This can be done with `ovs-ofctl dump-flows` command.
+
+- After the new `ovs-vswitchd` process starts in the new pod, the startup script will check
+  for the presence of the saved flows file. If it exists, it will restore the
+  flows using `ovs-ofctl` command. It will then remove the file.
+
+- Once the flows are restored, the startup script will set
+  `other_config:flow-restore-wait` back to `false` in the OVSDB. This will
+  allow the new `ovs-vswitchd` process to connect to its controllers and start
+  processing packets.
+
+## Bare metal Open vSwitch for data plane nodes
+
+In contrast to podified Open vSwitch for OVN gateways, Open vSwitch on data
+plane nodes is deployed on bare metal using RPM packages. The deployment is
+managed by `edpm_ovs` role.
+
+### RPM packaging
+
+There are several important details to consider when updating Open vSwitch RPM
+packages:
+
+- In addition to the `openvswitch*` package, OpenStack also ships a
+  `<distro>-openvswitch` wrapper package. (`rdo-openvswitch` for RDO,
+  `rhosp-openvswitch` for RHOSO). This package is used to select the Open
```

Comment:
nit: not sure if this is confirmed but the prefix will be changed to `rhoso` from `rhosp`

Reply:
Trying to figure out what's the final name for GA. For now leaving as-is, will update when I know for sure.

Reply:
It's `rhoso-*` (at least that's the plan for GA). The latest version pushed has this prefix documented.